### PR TITLE
Add stubs for services

### PR DESCRIPTION
**Pull request by lfdebrux**
We want stubs for

- [x] Services
- [x] DraftServices
- [x] ArchivedServices

Required to close https://trello.com/c/jEZzVo2u/326-tidy-signature-page-generation-script

### CODE REVIEW

File: dmtestutils/api_model_stubs/base.py [hunk 1]
```diff
@@ -27,13 +30,55 @@ def test_a_view_that_needs_briefs_from_the_api(mocked_get_brief_method):
...
-        self.response_data.update(**kwargs)
+    def _format_framework(self, slug, oldstyle=False):
+        """Return a dictionary with correct keys for framework slug"""
+        family, iteration = re.match(r"^(?P<family>[a-z-]*)(?:-(?P<iteration>\d+))?$", slug).groups()
+        name = (
+            {
+                "g-cloud": "G-Cloud",
+                "digital-outcomes-and-specialists": "Digital Outcomes and Specialists",
+            }.get(family,
+                  slug.translate(str.maketrans("-", " ")).title())
```
Comment (benvand) [inline dmtestutils/api_model_stubs/base.py]: Any benefit to `translate` over `replace`?

File: dmtestutils/api_model_stubs/base.py [hunk 2]
```diff
@@ -27,13 +30,55 @@ def test_a_view_that_needs_briefs_from_the_api(mocked_get_brief_method):
...
+        }
+        if not oldstyle:
+            return framework
+        else:
+            oldstyle_keys = {
+                "frameworkFamily": "family",
+                "frameworkSlug": "slug",
+                "frameworkName": "name",
+            }
+            return {oldstyle_key: value for oldstyle_key, value in zip(oldstyle_keys.keys(), framework.values())}
```
Comment (benvand) [inline dmtestutils/api_model_stubs/base.py]: Can use just `dict(zip(x, y))`here but comprehension is cool too, possibly better as more descriptive 👍

File: tests/test_api_model_stubs.py [hunk 3]
```diff
@@ -433,6 +516,51 @@ def test_returns_mapping_which_can_be_changed_using_kwargs(self, kwarg, key, val
         assert LotStub(**{kwarg: value}).response()[key] == value
 
 
+class TestServicesStubs:
```
Comment (benvand) [inline tests/test_api_model_stubs.py]: You can get rid of that `cls` fixture if you parametrize the class itself:
```
 
+@pytest.mark.parametrize('cls', [ArchivedServiceStub, DraftServiceStub, ServiceStub])
 class TestServicesStubs:
 
-    @pytest.fixture(params=[ArchivedServiceStub, DraftServiceStub, ServiceStub])
-    def cls(self, request):
-        return request.param
-
```

File: dmtestutils/api_model_stubs/base.py (hunk 1 above)
Comment (lfdebrux) [reply dmtestutils/api_model_stubs/base.py]: Nope, I can't believe I forgot about replace :man_facepalming:

File: dmtestutils/api_model_stubs/base.py (hunk 2 above)
Comment (lfdebrux) [reply dmtestutils/api_model_stubs/base.py]: Uh you can see I wasn't paying full attention yesterday... I mean, yes more descriptive, that's what I was going for!

File: tests/test_api_model_stubs.py (hunk 3 above)
Comment (lfdebrux) [reply tests/test_api_model_stubs.py]: I am just getting schooled today :laughing: Nice trick

File: dmtestutils/api_model_stubs/base.py (hunk 2 above)
Comment (lfdebrux) [reply dmtestutils/api_model_stubs/base.py]: Actually I think I remember why I started doing it as a comprehension, the idea was to not rely on the ordering of the dicts; obviously I forgot about this halfway through.

**Pull request merged**